### ISSUE TITLE
Removing items from the DOM

### ISSUE THREAD

**Issue by abbasarezoo** [patrickkunka/mixitup#292]
Bit of JS rookie but I've used the previous version of this plugin before and I'm pretty familiar with the config. I tend to use it on most projects.

I have a current project where I have a need for the ability to remove elements completely from the DOM when filtering, so that the remaining elements inherit specific CSS styles,  and vice-versa. Meaning the removed element would be injected back into the DOM. This might be completely unrealistic and unachievable but here goes...

I may have styles for grid elements like:

```css
.grid-item:nth-child(1) {
  background-color: red;
}

.grid-item:nth-child(2) {
  background-color: blue;
}
```
with this markup:

```html
<div class="grid">
  <div class="mix grid-item one"><h1>Grid item</h1></div>
  <div class="mix grid-item two"><h1>Grid item</h1></div>
</div>
```

So theoretically if filtered `<div class="mix grid-item one">` would be removed leaving the second one to take its place as `nth-child(1)` inheriting the `background-color: red;` styles. Then when all items a shown afterward `<div class="mix grid-item one">` goes back to its original spot and both elements take their original styles.

Hope that made sense, is this possible??

Here's a Codepen (http://codepen.io/abbasarezoo/live/77e615e505fbec78df4aaa1023a834af) to a similar prototype idea, with absolutely positioned elements. The ideal scenario would be that the remaining elements inherit positions and styles when others are filtered away.

Any help would be great!

**Comment by patrickkunka** [comment]
Hi @abbasarezoo,

Unfortunately nth-child based CSS and MixItUp (or any client side animated filtering) don't work together well. I wrote a bit about it under the "floats" section of this tutorial:

https://www.kunkalabs.com/tutorials/mixitup-grid-layouts/

Your proposed solution of removing the items from the DOM would work if you are happy to disable animations (but then you may not need MixItUp at all). The problem with animated filtering, is that any items to be added/removed have to exist in the DOM for the lifecycle of the animation and this means that you can't rely on the their CSS indices, and you will end up with an abrupt jump in styling at the beginning or end of the animation if using nth-child.

If you're happy to do away with animations, then I can advise further on ways to do filter/remove simultaneously.

**Comment by abbasarezoo** [comment]
Hi @patrickkunka - thanks for the speedy response. I thought that might be the case. Another idea had would be to add a data attribute to the remaining displayed items and style those instead?

So something like this after filtering:

```html
<div class="mix grid-item one" style="display: none;"></div>
<div class="mix grid-item two" data-order="1"></div>
<div class="mix grid-item three=" data-order="2"></div>
```
Or are we going over old ground here?

Either way, great plugin, thanks for the help!

**Comment by patrickkunka** [comment]
Yeah that's a great solution, (again if you don't mind some jumping of styling when you apply them at the end of the animation). I think I actually used this approach on a project a few years ago where we had a grid with alternating background colours on the items.

I had a special function which I called on `onMixEnd` that iterated through the `state.show` array. Depending on whether the item's index was even or odd (using modulus), a specific styling class was applied/removed.

Make sense?
